Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -8999,5 +9071,30 @@ private void VerticalScrollBar_Scroll(object sender, ScrollEventArgs e)
         {
             ProcessVerticalScroll(e.ScrollEventType);
         }
+
+        private bool _loadingItems;
+
+        private async void LoadMoreDataFromIncrementalItemsSourceAsync()
+        {
+            if (IncrementalLoadingTrigger == IncrementalLoadingTrigger.Edge && DataConnection.IsDataSourceIncremental && DataConnection.HasMoreItems && !_loadingItems)
```

Comment:
You can replace "&& !_loadingItems" with "&& !DataConnection.IsLoadingMoreItems".